src/Framework/TestCase.php

```diff
@@ -174,6 +179,10 @@ protected function tearDown()
 	{
 	}
 
+	/** @internal */
+	public function failsafeTearDown() {
+		$this->tearDown();
```

Comment:
@milo well, if tearDown() fails, there is nothing more we can do (point of this pull request is to execute tearDown if setUp or test* fails).